src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -429,8 +441,10 @@ public async Task DispatchEvent(string eventDescriptorJson, string eventArgs)
             }
             catch (Exception ex)
             {
+                // An error dispatching an is not considered fatal.
+                // Usually this kind of error comes from the renderer (unknown event id) not from user code.
+                // Exceptions from user code come through the renderer's unhandled exception event.
```

Comment:
I think @SteveSandersonMS wants an error dispatching an event to turn down the circuit, as its a signal that a client is misbehaving, so we want to turn the circuit down.

Reply:
Yes, an unknown event ID should be fatal to the circuit. It's not supposed to be possible for a well-behaved client, even allowing for asynchrony. The renderer won't dispose event handlers until it knows the client has acknowledged that event handler is no longer going to be used, and the ordering of messages ensures we won't later get any message referring to that ID.

Reply:
So one of the things that I did here was make sure that the existing tests in this area pass. I didn't scrutinize the behavior of the existing tests or logic. 

Isn't this **specific** default covered by another PR you have in flight? I'd really like to not increase the scope of this single PR more than necessary.

Reply:
> So one of the things that I did here was make sure that the existing tests in this area pass.

We wouldn't have had a test for this before. The absence of this functionality before is an aspect of the bug https://github.com/aspnet/AspNetCore/issues/11845 that I thought this PR was addressing (assuming you want to address that aspect of it).

Previously, if event dispatch threw, then `CircuitHost` would fire its `UnhandledException` event which made `ComponentHub` notify the client, and trust that the client would disconnect. My intent with filing #11845 was to say that in general, the `UnhandledException` code path should force the client to disconnect and not trust that it will do so.

The actual writeup in #11845 predates Javier's work to convert events-over-JS-interop into events-as-first-class, so the notes there don't exactly reflect the current state.

> Isn't this specific default covered by another PR you have in flight?

Do you mean https://github.com/aspnet/AspNetCore/pull/12683/files? Unless I'm misreading, PR #12683 only ensures we don't log the invalid event ID as warning (since bad IDs can be sent arbitrarily by misbehaving clients), but it doesn't change anything about forcing a disconnect.

> I'd really like to not increase the scope of this single PR more than necessary.

Totally understandable. If you prefer we handle that separately we can. It does seem very closely related to the subject matter for this PR though.

Reply:
Actually I've now checked:

 * On `release/3.0`, sending an invalid event payload sends an error back to the browser console, and for a well-behaved client, means we terminate the connection.
 * With this PR, sending an invalid event payload is totally silent (assuming you're not logging at debug level), and the connection keeps going. That's not ideal.

I definitely think we should address this as part of preview 9, whether or not it's in this PR.